apollo-router/src/services/router_service.rs

```diff
@@ -228,119 +229,125 @@ impl RouterService {
             }
         };
 
-        let mut request_res = self
-            .persisted_query_layer
-            .supergraph_request(supergraph_request);
+        let mut results = vec![];
 
-        if let Ok(supergraph_request) = request_res {
-            request_res = self.apq_layer.supergraph_request(supergraph_request).await;
-        }
+        for supergraph_request in supergraph_requests {
+            let mut request_res = self
```

Comment:
split the loop body in a separate function maybe?

Reply:
Yeah, the formatter has given up on this completely. I'll do some tidying up at some point. Just making the basic code work at the moment.